river/facto/base.py

```diff
@@ -122,9 +125,103 @@ def _field(self, j):
         """Infers feature field name."""
         return j.split("_")[0]
 
-    @abc.abstractmethod
     def _calculate_interactions(self, x: dict) -> float:
-        """Calculates greater than unary interactions."""
+        """Calculates greater than unary interactions.
+        For normal FM: sigma (i < j) product(latents[xi] * latents[xj]) xi * xj
+        interaction_combination_keys: sigma (i < j)
+        interaction_val: xi * xj
+        interaction_coefficient: latents[xi] * latents[xj]
+        """
+        return sum(
+            [
+                self._interaction_coefficient(combination)
+                * self._interaction_val(x, combination)
+                for combination in self._interaction_combination_keys(x)
+            ]
```

Comment:
You don't need the `[` and `]` here

Reply:
Fixed in https://github.com/online-ml/river/pull/787/commits/20ae38f372a3aca1a44b8a17b60076b3cb6e9de7